rountine-channel.go

```diff
@@ -0,0 +1,99 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"strings"
+	"sync"
+	"time"
+)
+
+var fileDir = "./data"
+var dict map[string]int
+var mapMutex = sync.RWMutex{}
+
+//Job ...
+type Job struct {
+	id       int
+	fileName string
+}
+
+//MaxWorker ...
+const MaxWorker int = 10
+
+var jobs = make(chan Job, MaxWorker)
+
+func countWordsFromFile(fileName string) {
+	st, err := ioutil.ReadFile(fmt.Sprintf("%s/%s", fileDir, fileName))
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	words := strings.Fields(string(st))
+
+	for i := 0; i < len(words); i++ {
+		mapMutex.Lock()
```

Comment:
Another approach is that all workers send their results to a channel and another goroutine will aggregate data.